backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ObservationHandler.java

```diff
@@ -225,11 +226,11 @@ public ObservationHandler(Executor executor, FutureJdbi jdbi) {
         handle -> {
           // Delete from keyvalue
           var sql =
-              "delete from keyvalue where id in "
-                  + "(select keyvaluemapping_id from observation where entity_name=\"ExperimentRunEntity\" and field_type=\"observations\" and experiment_run_id=:run_id)";
+              "delete o, k from keyvalue AS k inner join observation AS o ON o.keyvaluemapping_id=k.id "
+                  + "where o.entity_name=\"ExperimentRunEntity\" and o.field_type=\"observations\" and o.experiment_run_id=:run_id ";
```

Comment:
In the old query we can not delete the keyvalue because there is a foreign key constraint with observation. so removed implementation of query doesn't work. So added new query after testing and new query is working fine now.